### PR TITLE
feat: check enabled tests task

### LINKED ISSUE #799: [Spec tests] Coverage script
This script will not measure the coverage of the code, but the spec test coverage, with a detailed report, either in CLI or html format.

Currently, implemented and skipped tests are clearly output when running make spec-test, but we don't have any measure of unimplemented runners. We should have a script that checks:

- Which test vectors have runners implemented.
- Which test cases are skipped vs pass.

This would allow us to not miss any spec tests because of test runners not being implemented.

Some configs could be:

- Filters:
  - Per config (mainnet, minimal, general).
  - Per fork.
  - Per runner.
- Show only unimplemented.

### PR DESCRIPTION
**Pull request by Arkenan**
Changes:

- `check_enabled_tests` task that generates a report on skipped vs enabled tests.
- New makefile target to run iex in test mode.
- MetaUtils module for code inspection.

Closes #799

### CODE REVIEW

File: test/spec/tasks/check_enabled_tests.ex [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  require Logger
+
+  alias Spec.MetaUtils
+  @output_dir "reports"
+
+  def run(_args) do
+    # TODO: pretty print.
+    res = MetaUtils.check_enabled() |> inspect(limit: :infinity)
+    File.mkdir_p!(@output_dir)
+    File.write!(filename(), res)
```
Comment (MegaRedHand) [inline test/spec/tasks/check_enabled_tests.ex:21]: I think we should just `IO.puts` this, for simplicity.

Comment (Arkenan) [reply test/spec/tasks/check_enabled_tests.ex]: Removed the file, added pretty printing.

File: test/spec/tasks/check_enabled_tests.ex [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  alias Spec.MetaUtils
+
+  def run(_args) do
+    MetaUtils.check_enabled() |> print(0)
+    :ok
+  end
+
+  defp print({name, report}, spaces) do
+    print_spaced(name, spaces)
+    new_spaces = spaces + String.length(name)
```
Comment (MegaRedHand) [inline test/spec/tasks/check_enabled_tests.ex:23]: To account for long names, I think it's better to use a fixed size here.

```elixir
    new_spaces = spaces + 4  # may be a module attribute
```

File: test/spec/tasks/check_enabled_tests.ex [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  Run with MIX_ENV=test mix check_enabled_tests
+  The output will be in a reports/spec_test_report.exs file, that can be formatted using mix format.
+  """
+
+  require Logger
+
+  alias Spec.MetaUtils
+
+  def run(_args) do
```
Comment (MegaRedHand) [inline test/spec/tasks/check_enabled_tests.ex:16]: I haven't been able to run this from the CLI (tried with `mix check_enabled_tests`). I noticed it's missing the usual `use Mix.Task` and `@impl Mix.Task` tags, but even with that it doesn't work. Is it working for you?

**Review by MegaRedHand — CHANGES_REQUESTED** [review]

File: test/spec/tasks/check_enabled_tests.ex (hunk 2 above)
Comment (Arkenan) [reply test/spec/tasks/check_enabled_tests.ex]: Added 3-space indents + 🟡 emoji for the partial runs. This way, the partial runs are aligned with other titles and lines are not that long.

File: test/spec/tasks/check_enabled_tests.ex (hunk 3 above)
Comment (Arkenan) [reply test/spec/tasks/check_enabled_tests.ex:20]: Fixed. You had to use MIX_ENV=test, but I added the env preference so you can skip the manual env config.

**Review by MegaRedHand — APPROVED** [review]
Great work!! 🔥

**Pull request merged**